specs/flyout.md

```diff
@@ -0,0 +1,152 @@
+# Flyout
+
+## Overview
+Flyout is a focusable floating container that shows over a pages content and displays UI related to what the user is doing. Similar to a Dialog, but with positioning controls and no specific action is required, a Flyout can be used to reveal a secondary control or show more detail about an item.
+
+### Background
+A Flyout can be less intrusive to the user experience than a Dialog and should be used instead of a Dialog when information or input is needed to be displayed, but not in the center of the screen, such as in the case of collection of additional info before an action, displaying information that is only relevant some of the time, or displaying more information.
+
+### Use Cases
+- As a Dropdown for a menu, navigation bar, etc.
+- In a context menu as the secondary menu/action.
+- As a popup showing additional info for an item on the page.
+- To display a larger image of a thumbnail image in a photo gallery.
+- To display an action related to a page item
+
+### Features
+**Placement Logic:** - This component will use *Anchored Region* to be positioned relative to the Flyouts anchor element. This way it can be dynamic and adapt to the available space if `responsive` is true for the Flyout.
+
+**Light Dismiss (aka Soft Dismiss):** - A way to quickly close the Flyout by clicking outside it, or pressing the `esc` key.
+
+**Overlay:** - A control of whether or not the Flyout has an overlay effect that dims the surrounding UI when the Flyout is active.
+
+### Prior Art/Examples
+##### Context Menu
+*Note:* the "Manage, Apply, Show more options" is the Flyout for this example.
+![Context Menu](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-nested.png)
+
+##### Text Only
+*Note:* the button is not part of the Flyout, but shown for context as an anchor element.
+![Text Only](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-wrapping-text.png)
+
+##### Actionable
+*Note:* the button is not part of the Flyout, but shown for context as an anchor element.
+![Actionable More Info](https://docs.microsoft.com/en-us/windows/uwp/design/controls-and-patterns/images/flyout-example2.png)
+
+---
+
+### API
+*Component Name*
+- `fast-flyout`
+
+*Attributes*
+- `anchor` - string, the html id of the HTMLElement that the Flyout is attached to
+- `overlay` - boolean, specifies whether an overlay is drawn to dim the out of scope UI until the Flyout is dismissed
+- `light-dismiss` - boolean, specifies whether or not the flyout can be dismissed by clicking elsewhere or pressing `esc` key, defaults to true
+- `visible` - boolean, whether or not the Flyout is visible, defaults to undefined, as this is controlled by the author
+- `responsive` - boolean, whether or not the positioning is responsive based on available space, defaults to true, uses Anchored Region logic
```

Comment:
How does "responsive" interact with "position"?  Does it affect both axis?  Is a single bool enough to cover the "responsive" behavior we want?  For example "I want my flyout to be above or below the anchor" and "I want my flyout to be left or right of the anchor" seem like reasonable choices.  Not sure how much complexity do we want here but we probably need more than a single bool?  Note that in tooltip we currently default to a "above or below based on available space" default placement if no fixed position is provided.  We could do something similar here, skip the responsive prop and avoid having to define a that part of the api now and just pick the default behavior.

Reply:
I see what you're saying. I was going for a more general position change. `responsive` set to true would allow the flyout to change it's location if the location the author specified doesn't have available space. Having `responsive` set to false would NOT allow the flyout to move if there is not enough space in the position the author has specified.
I could see some authors NOT wanting the flyout to move at all based on available space and want to give them the option to specify if it is "fixed" vs "responsive" with regards to available space. Does that make sense with Anchored Region? I was under the impression that we could setup anchored region so that it does not change its location based on available space.

Reply:
Why not do this with the position attribute?  If an author sets the position to "top" it goes on top. For the first release if position is "undefined" we do "above or below based on space"?  That leaves us free to evolve a more complex placement system later.